synapse/http/server.py

```diff
@@ -257,7 +257,7 @@ async def _async_render(self, request: Request):
             if isinstance(raw_callback_return, (defer.Deferred, types.CoroutineType)):
                 callback_return = await raw_callback_return
             else:
-                callback_return = raw_callback_return
+                callback_return = raw_callback_return  # type: ignore
```

Comment:
I could not figure out how to coerce mypy into realize that this might be an awaitable or might not be. 🤷